lib/federation.js

```diff
@@ -0,0 +1,210 @@
+'use strict'
+
+const {
+  buildSchema,
+  extendSchema,
+  parse,
+  isObjectType,
+  Kind,
+  isTypeDefinitionNode,
+  isTypeExtensionNode
+} = require('graphql')
+
+const federationSchema = `
+scalar _Any
+scalar _FieldSet
+
+# a union of all types that use the @key directive
+# union _Entity
+
+type _Service {
+  sdl: String
+}
+
+type Query {
+  #_entities(representations: [_Any!]!): [_Entity]!
+  _service: _Service!
+}
+
+directive @external on FIELD_DEFINITION
+directive @requires(fields: _FieldSet!) on FIELD_DEFINITION
+directive @provides(fields: _FieldSet!) on FIELD_DEFINITION
+directive @key(fields: _FieldSet!) on OBJECT | INTERFACE
+
+# this is an optional directive discussed below
+# directive @extends on OBJECT | INTERFACE
+`
+
+function hasDirectives (node) {
+  return Boolean('directives' in node && node.directives)
+}
+
+function gatherDirectives (type) {
+  let directives = []
+  if ('extensionASTNodes' in type && type.extensionASTNodes) {
```

Comment:
Please use hasOwnProperty instead

Reply:
If I use that the lint fails (`Do not access Object.prototype method 'hasOwnProperty' from target object`)

Reply:
Yes, you need to get the function from the prototype and use `call()`.